### DISCUSSION TITLE
Did not find matching provisioning profiles for code signing!

### DISCUSSION THREAD

**Discussion by noorbhatia** [codemagic-ci-cd/codemagic-docs#2492] [Q&A]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

iOS (Swift, Objective-C)

### Steps to reproduce

scripts:
 - name: Set up provisioning profiles settings on Xcode project
        script: xcode-project use-profiles

### Expected results

find the correct ios profile

### Actual results

```Configure code signing settings
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.mobileprovision
Searching for files matching /Users/builder/Library/MobileDevice/Provisioning Profiles/*.provisionprofile
List available code signing certificates in keychain /Users/builder/Library/codemagic-cli-tools/keychains/23-04-24_3sqzgqtl.keychain-db
Searching for files matching /Users/builder/clone/Hunbun/**/*.xcodeproj
Completed configuring code signing settings
Did not find matching provisioning profiles for code signing!
Generated options for exporting the project
 - Method: ad-hoc
 - Provisioning Profiles: []
 - Signing Certificate: 
 - Signing Style: manual
 - Team Id: 
Saved export options to /Users/builder/export_options.plist
```

### Build id (optional)

66281d83a4e02c8e34b09c8c

**Comment by noorbhatia** [comment]
I have tried with new matching profiles and certificates but it's still failing

buildId: 6628f7cfbd7a34bd54b8a4d8

@dtrdic can you help?

**Reply by dtrdic** [reply]
hi @noorbhatia i can see this error inside logs
`error: Bundle identifier is missing. Hunbun doesn't have a bundle identifier. Add a value for PRODUCT_BUNDLE_IDENTIFIER in the build settings editor`

Please open your iOS project in an editor and open the` project.pbxpro`j file and set the application `PRODUCT_BUNDLE_IDENTIFIER` . This is mentioned in multiple places in the project, so make sure to change it everywhere.
Also add the archive flags attribute to the end of your build command
```
xcode-project build-ipa \
  --workspace "$CM_BUILD_DIR/Hunbun/$XCODE_WORKSPACE" \
  --scheme "$XCODE_SCHEME" \
  --archive-flags="-destination 'generic/platform=iOS'"
```

**Reply by noorbhatia** [reply]
@dtrdic , the bundle identifier is picked from custom config files
<img width="398" alt="image" src="https://github.com/codemagic-ci-cd/codemagic-docs/assets/73331825/090225b6-689d-453d-846f-929f52690085">
I have three configs; dev, staging and prod. What can I do?

**Reply by noorbhatia** [reply]
self resolved

**Reply by usumfabricae** [reply]
How did you fix it?

**Reply by micheljung** [reply]
@noorbhatia mind telling us how you fixed it?

**Reply by husshd93** [reply]
> self resolved

Do you still remember how you solved it?

**Reply by dtrdic** [reply]
@husshd93 please open a new ticket and share a build id of your failing build and somebody from the Codemagic team will check out your issue
